core/import-csv.c

```diff
@@ -0,0 +1,944 @@
+		break;
+	case POSEIDON_SENSOR2:
+		sample->o2sensor[1].mbar = lrint(val * 10);
+		break;
+	case POSEIDON_NDL:
+		sample->ndl.seconds = lrint(val * 60);
+		break;
+	case POSEIDON_CEILING:
+		sample->stopdepth.mm = lrint(val * 1000);
+		break;
+	}
+}
+
+char *parse_dan_new_line(char *buf, const char *NL)
+{
+	char *iter = buf;
+
+	if (!iter)
+		return NULL;
+
+	iter = strstr(iter, NL);
+	if (iter) {
+		iter += strlen(NL);
+	} else {
+		fprintf(stderr, "DEBUG: No new line found");
+		return NULL;
+	}
+	return iter;
+}
+
+int parse_dan_format(const char *filename, char **params, int pnr)
+{
+	int ret = 0, i;
+	size_t end_ptr = 0;
+	struct memblock mem, mem_csv;
+	char tmpbuf[MAXCOLDIGITS];
+
+	char *ptr = NULL;
+	char *NL = NULL;
+	char *iter = NULL;
+
+	if (readfile(filename, &mem) < 0)
+		return report_error(translate("gettextFromC", "Failed to read '%s'"), filename);
+
+	/* Determine NL (new line) character and the start of CSV data */
+	if ((ptr = strstr(mem.buffer, "\r\n")) != NULL) {
+		NL = "\r\n";
+	} else if ((ptr = strstr(mem.buffer, "\n")) != NULL) {
+		NL = "\n";
+	} else {
+		fprintf(stderr, "DEBUG: failed to detect NL\n");
+		return -1;
+	}
+
+	while ((end_ptr < mem.size) && (ptr = strstr(mem.buffer + end_ptr, "ZDH"))) {
+		char *iter_end = NULL;
+		unsigned int pnr_local = pnr;
+
+		/*
+		 * Process the dives, but let the last round be parsed
+		 * from C++ code
+		 */
+
+		if (end_ptr)
+			process_dives(true, false);
+
+		mem_csv.buffer = malloc(mem.size + 1);
+		mem_csv.size = mem.size;
+
+		iter = ptr + 4;
+		iter = strchr(iter, '|');
+		if (iter) {
+			memcpy(tmpbuf, ptr + 4, iter - ptr - 4);
+			tmpbuf[iter - ptr - 4] = 0;
+			params[pnr_local++] = "diveNro";
+			params[pnr_local++] = strdup(tmpbuf);
+		}
+
+		//fprintf(stderr, "DEBUG: BEGIN end_ptr %d round %d <%s>\n", end_ptr, j++, ptr);
+		iter = ptr + 1;
+		for (i = 0; i <= 4 && iter; ++i) {
+			iter = strchr(iter, '|');
+			if (iter)
+				++iter;
+		}
+
+		if (!iter) {
+			fprintf(stderr, "DEBUG: Data corrupt");
+			return -1;
+		}
+
+		/* Setting date */
+		memcpy(tmpbuf, iter, 8);
+		tmpbuf[8] = 0;
+		params[pnr_local++] = "date";
+		params[pnr_local++] = strdup(tmpbuf);
+
+		/* Setting time, gotta prepend it with 1 to
+		 * avoid octal parsing (this is stripped out in
+		 * XSLT */
+		tmpbuf[0] = '1';
+		memcpy(tmpbuf + 1, iter + 8, 6);
+		tmpbuf[7] = 0;
+		params[pnr_local++] = "time";
+		params[pnr_local++] = strdup(tmpbuf);
+
+		/* Air temperature */
+		memset(tmpbuf, 0, sizeof(tmpbuf));
+		iter = strchr(iter, '|');
+
+		if (iter && iter + 1) {
+			iter = iter + 1;
+			iter_end = strchr(iter, '|');
+
+			if (iter_end) {
+				memcpy(tmpbuf, iter, iter_end - iter);
+				params[pnr_local++] = "airTemp";
+				params[pnr_local++] = strdup(tmpbuf);
+			}
+		}
+		params[pnr_local] = NULL;
+
+		/* Search for the next line */
+		if (iter)
+			iter = parse_dan_new_line(iter, NL);
+		if (!iter) {
+			fprintf(stderr, "DEBUG: No new line found");
```

Comment:
missing `\n` ?

Reply:
Fixing the \n in new_line function and removing these prints as it is sufficient to have the debug string printed only once.